deku-derive/src/macros/mod.rs

```diff
@@ -65,3 +59,95 @@ fn gen_struct_init<I: ToTokens>(is_named: bool, field_idents: Vec<I>) -> TokenSt
         }
     }
 }
+
+/// Generate struct destruction
+/// Named: `#ident { ref fields }`
+/// Unnamed: `#ident ( ref fields )`
+fn gen_struct_destruction<I: ToTokens, F: ToTokens>(
+    named: bool,
+    ident: I,
+    field_idents: &[F],
+) -> TokenStream {
+    if named {
+        quote! {
+            #ident {
+                #(ref #field_idents),*
+            }
+        }
+    } else {
+        quote! {
+            #ident (
+                #(ref #field_idents),*
+            )
+        }
+    }
+}
+
+fn gen_hidden_field_ident(ident: TokenStream) -> TokenStream {
+    // We can't concat to token, so I use string.
+    // See https://github.com/rust-lang/rust/issues/29599
+    let span = ident.span();
+    let s = ident.to_string();
+    let mut name = "__".to_owned();
```

Comment:
we could rename `hidden` to `internal` and change `__` to `__deku_` to imply internal variable

Reply:
Ah I didn't know it before. Thanks